### DISCUSSION TITLE
difference in standard "pacman" st and compiled st

### DISCUSSION THREAD

**Discussion by rayvermey** [bakkeby/st-flexipatch#38] [Q&A]
Hi

when i start the normal via pacman on Arch installed st, my prompt is like in file a
When i use the st-flexipatch compiled version i have a prompt like in file b.
Any idea how i can have it like file a
![a](https://user-images.githubusercontent.com/42524768/127938279-b38f8fb9-7095-484d-a82c-3ae42d0987e9.png)
![b](https://user-images.githubusercontent.com/42524768/127938281-e9da6ce8-6779-41d1-b977-def5119bc1b3.png)
?
Thanks.

**Comment by bakkeby** [comment]
For these kind of powerline symbols you will need a font that has those glyphs, e.g. Font Awesome or Nerd Font  variants.

Check the font configuration in config.h.

https://github.com/bakkeby/st-flexipatch/blob/d50ebeef66cde93d440d027d0459d9fa3a923e33/config.def.h#L8

You may also want to have a look at the font2 patch to be able to add fallback fonts.

**Reply by rayvermey** [reply]
Yes i did, thanks for pointing that out BUT whatever font i choose, it changes yesm but not the way i want it, the blue round arrow so to speak just does not appear... I will keep trying though,
Is there a way to discover which font the /usr/bin/st version uses?

**Reply by bakkeby** [reply]
Try installing gucharmap, copy the round arrow character from the st where it displays correctly,  then do a search for that character in gucharmap. Right clicking on the character should give you a larger preview and more information like what font it comes from.

**Reply by rayvermey** [reply]
Yes works now! Had to try more fonts and finaly had one which displays fine, THANKS again for your help
I'll buy you a coffee ok? :-)

**Reply by rayvermey** [reply]
I have a new strange thing and i hope i can ask it here (btw how can i buy you that coffee? ;-)  )
If i change the font for the statusbar (the bar on top where the tags are) then all of a sudden the bar is gone, well not
totally gone but hiding behind the wallpaper....
WHY??? Is there a specific font i should use??

**Reply by bakkeby** [reply]
I hope you are referring to dwm, not that you now have a bar with tags in st...

What does your font config look like?

The wallpaper is drawn on the root window and nothing can be behind that. If you only see the wallpaper where the bar should be then it means that the bar window is not there (maybe the window failed to be created).

If I were to guess then I would say that you are not using a valid font or not specifying the font size correctly.

**Reply by rayvermey** [reply]
Hi, yes sorry for talking about DWM in a ST thread.. so much for netiquette :)
Yes there is a small line with my bar, i can click there for the tags if i position my mousepointer with care.
Indeed it was when i used several fonts that the bar returned.
That makes me wonder "how can i know which font i can use for the bar??"
About the flexi-patch stuff: I have used the finalizer script which gives me a nice looking config.
Can i then later edit the patches.def.h again to switch on other patches? Or should i work with the original git clone and
finalize again?
Thanks for you patience and support! Much appreciated!

**Reply by bakkeby** [reply]
Can you show me your fonts config item?

If you finalize then you can keep a backup of the source directory which you can use to make amendments. 

You may want to make the config changes manually so as to not overwrite any if your settings.

**Reply by rayvermey** [reply]
You mean this?
 static const char font[]                 = "Noto Sans 14";
 30 static const char dmenufont[]            = "-*-terminus-medium-r-normal-*-16-*-*-*-*-*-*-*";

OK, so i finalize, and then adjust my config along the way, if i later want to add other patches, then i do that in the source
directory, finalize again and diff my configs? **puzzled** , but just woke op so maybe i am still a little dense :-)

**Reply by bakkeby** [reply]
Oh so you are using the Pango patch?

You can also use the bar_height patch to set the height of the bar absolutely (i.e. the bar height won't be derived from the font height).

The finalize bit is, you know, to make it final, as in no further changes. If you are still playing around trying out patches then there is not really much point in finalizing the build.
